services/thermal_mgr/thermal_mgr.c

```diff
@@ -21,51 +21,100 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
 static QueueHandle_t thermalMgrQueueHandle;
 static StaticQueue_t thermalMgrQueueBuffer;
 static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
-
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_UNKNOWN;
+
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *)event, (TickType_t)0) == errQUEUE_FULL)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
-  /* Implement this function */
+void osHandlerLM75BD(void)
+{
+
+  float temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+
+  if (temp >= LM75BD_DEFAULT_HYST_THRESH)
+    overTemperatureDetected();
+
+  else if (temp <= LM75BD_DEFAULT_HYST_THRESH)
+    safeOperatingConditions();
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters)
+{
+
+  float temp = 0;
```

Comment:
In general, limit the scope of variables as much as possible. You can limit the scope of `temp`.